### PR TITLE
sensors: add ADXRS290 driver

### PR DESCRIPTION
**Pull request by MaxAgneni**
Signed-off-by: Massimiliano Agneni <massimiliano.agneni@iptronix.com>

### CODE REVIEW

File: drivers/sensor/adxrs290/adxrs290.c [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct adxrs290_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+		/* Angular velocity around the X axis, in radians/s. */
+		val->val1 = (s32_t)data->x / ADXRS290_GYR_SCALE_FACT;
+		val->val2 = ((s32_t)data->x % ADXRS290_GYR_SCALE_FACT)
```
Comment (bogdan-davidoaia) [inline drivers/sensor/adxrs290/adxrs290.c]: You still need to divide val2 with ADXRS290_GYR_SCALE_FACT. This applies to GYRO_Y and TEMP as well.

File: drivers/sensor/adxrs290/adxrs290.c [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+		return ret;
+	}
+
+	data = lsb;
+	data |= (0xFF & msb) << 8;
+
+	/* CONVERT 12 bit, into 2 complement */
+	*temp = (s16_t)data;
+	/* Decode Temp measurement */
+	*temp /= ADXRS290_TEMP_SCALE_FACT;
```
Comment (bogdan-davidoaia) [inline drivers/sensor/adxrs290/adxrs290.c]: Why divide here? Isn't it also done in the channel_get function?

**Review by bogdan-davidoaia — CHANGES_REQUESTED** [review]
Sorry for the late reply. I believe Shippable was showing errors on this and I wasn't notified of any changes in the patch, so I overlooked it.

**Comment by nashif** [comment]
recheck

**Comment by tbursztyka** [comment]
This will require to use new spi API. (But I opened an issue about sensor drivers that need that change, so I will not block that PR for this)

**Comment by nashif** [comment]
stale PR

**Pull request closed without merging**